### ISSUE TITLE
Release for Elm 0.19?

### ISSUE THREAD

**Issue by michaeljones** [rluiten/elm-date-extra#71]
We're very happily using this library at work. I don't believe it has been released for Elm 0.19 yet. Could we help with that at all? Are there any blocking issues that you're aware of? I'm guessing `unsafeFromString` from string might be an issue as it is my understanding that `Debug.crash` is no longer available?

**Comment by rluiten** [comment]
I havn't looked at 0.19 yet, I did notice it was released.
I hope to allocate some time next weekend.

For Debug.crash I suppose just returning a date maybe something like unix epoch 1970 instead would at least let the Api not change. Even if in my opinion its not as useful as a crash for something called `unsafedFromString`  :).

**Comment by michaeljones** [comment]
Yeah, I wasn't against `crash` being removed but I'm using that particular method in the tests and have no problem with it there at all.

I've looked a little closer and 0.19 removes the core `Date` module to my surprise. Instead suggesting use of `elm/time`, but that doesn't provide a `Date` type so some significant changes are going to be needed I'm afraid. The other date extra library seem to be deciding that perhaps it doesn't make sense to update to 0.19 and instead to provide other libraries with different slices of functionality: https://github.com/justinmimbs/elm-date-extra/issues/9#issuecomment-417536383

I'm afraid this has rather caught me off guard and made me reassess the whole timeline for an Elm 0.19 update for our application code.

I would be happy to provide any assistance that I can to your project for the upgrade though if you do want to go ahead. We have found it very useful. We're mainly using `Compare` and maybe `Format` but some other stuff too.

**Comment by AdrianRibao** [comment]
I'd like to contribute to this package and try to make it work with elm 0.19, but I have never contributed an elm package earlier, and there is not good documentation out there.

It would be nice to add a [CONTRIBUTING](https://help.github.com/articles/setting-guidelines-for-repository-contributors/) file.

Basically I'd like to know how to clone, and link the project to an existing project, test it and commit.

**Comment by michaeljones** [comment]
I've switched over 0.19 and I've just ended up moving to other packages. The core Date & Time stuff has changed a lot in 0.19 and I think it is completely understandable if this package doesn't make the transition. Other packages are available now that manage a lot of the same problems. Many thanks to @rluiten for making this in the first place. It certainly has been very useful!

**Comment by rluiten** [comment]
I've done some research, and there appears to be a lot of work to try and migrate this package to 0.19, and it would not be the same once there anyway due to platform changes as you point out.

Possibly extracting some parts might be useful, but I haven't had a lot of time to work on Elm stuff lately.
So would like to apologise for my slow response.

**Comment by AdrianRibao** [comment]
I forked the project and try to migrate to 0.19, but the changes are so deep that I thought it would be easier start from scratch. That's what I did, I started https://github.com/AdrianRibao/elm-derberos-date

This is mi very first contribution, and information is scarce so I'll try to do my best with it.

All the operations in my library are being done using the serialized Posix time. I want to add a few calendar functions before releasing the first version, and introduce the timezone in the functions, but I first need an answer on this: https://github.com/elm/time/issues/8

**Comment by CoderDennis** [comment]
I published a fork for Elm 0.19. It strips out everything except for i18n Configs and one `format` function. The other parts of this library are covered well by other existing Elm 0.19 packages, but this felt like the best way to keep using the internationalization and formatting.

https://package.elm-lang.org/packages/CoderDennis/elm-time-format/latest/

@rluiten please let me know if there's anything else you would expect me to do regarding copyright and changing of the license. I preserved what was there, but I haven't published a fork like this before, so if I've done anything inappropriate, I'd like to know. Thanks!

**Comment by rluiten** [comment]
That actually looks like a good way to retain some of the value, I havn't spent much time in Elm in last few years only updating specific libraries or for Elm version changes so I am happy to see this go forward and still be useful.

**Comment by ad-si** [comment]
If someone still wants to upgrade this to 0.19, I basically already did it here: 
https://github.com/ad-si/vectual/tree/master/src/TimeUtils
